src/main/java/io/vertx/core/file/impl/AsyncFileImpl.java

```diff
@@ -459,16 +459,19 @@ private void checkContext() {
   }
 
   private void doClose(Handler<AsyncResult<Void>> handler) {
-    Future<Void> res = Future.future();
-    try {
-      ch.close();
-      res.complete(null);
-    } catch (IOException e) {
-      res.fail(e);
-    }
-    if (handler != null) {
-      vertx.runOnContext(v -> handler.handle(res));
-    }
+    ContextImpl handlerContext = vertx.getOrCreateContext();
+    context.executeBlocking((Future<Void> res) -> {
```

Comment:
Done

2016-08-18 14:44 GMT+02:00 Julien Viet <notifications@github.com>:

> In src/main/java/io/vertx/core/file/impl/AsyncFileImpl.java
> <https://github.com/eclipse/vert.x/pull/1574#discussion_r75300316>:
>
> > @@ -459,16 +459,19 @@ private void checkContext() {
> >    }
> >
> >    private void doClose(Handler<AsyncResult<Void>> handler) {
> > -    Future<Void> res = Future.future();
> > -    try {
> > -      ch.close();
> > -      res.complete(null);
> > -    } catch (IOException e) {
> > -      res.fail(e);
> > -    }
> > -    if (handler != null) {
> > -      vertx.runOnContext(v -> handler.handle(res));
> > -    }
> > +    ContextImpl handlerContext = vertx.getOrCreateContext();
> > +    context.executeBlocking((Future<Void> res) -> {
>
> execute on handlerContext directly and pass the handler as resultHandler
> of executeBlocking
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/vert.x/pull/1574/files/d6bb6ef0fb00577c1aa348bbc63bd90b71680121#r75300316>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbltmcQvWCs5pKEc-HkEVYLxVZK5NVTks5qhFORgaJpZM4Jmmkg>
> .
>